specs/consensus/overview.md

```diff
@@ -492,13 +492,37 @@ selection algorithm becomes more challenging.
 
 ### Proposal value
 
-The external function `getValue()` is invoked by the proposer of a round as
-part of the transition to the [propose round step](#propose).
-It should return a value to be proposed by the process `p` for the current
-height of consensus `h_p`.
-
-> TODO: synchronous/asynchronous implementations, currently discussed
-> [here](../english/consensus/README.md#asynchronous-getvalue-and-proposevaluev).
+The `getValue()` function  is invoked by the [proposer](#proposer-selection) of
+a round when it is free to produce a new value to be proposed, namely in the
+first round of a height or when `validValue_p = nil` (more details in the
+[Value Selection](#value-selection) section).
+
+In other words, the `getValue()` function is the way in which Tendermint
+implements the **propose** consensus primitive.
+Instead of having the context/application that uses the consensus protocol
+invoking it with a value to be proposed, when a process running Tendermint can
+propose a value, it invokes this function so that the context/application can
+provide the value to be proposed.
+
+> Notice that this is a design concern that is encoded in Tendermint's
+> pseudo-code.
+> A priori, every process running a consensus height could provide its proposed
+> value for that height, but this approach has some disadvantages:
```

Comment:
I wonder whether we should add a bullet point on the Starknet usecase, where it is expensive to compute a proposal (if it must contain a ZK proof), and therefore only the scheduled proposers of round 0 should do that to optimize on the normal case.

Reply:
Added a bullet point with this example.